include/s3select.h

```diff
@@ -512,6 +524,10 @@ struct s3select : public bsc::grammar<s3select>
 
       data_type = (bsc::str_p("int") | bsc::str_p("float") | bsc::str_p("string") |  bsc::str_p("timestamp") );
 
+      trim = (bsc::str_p("trim") >> '(' >> (trim_type)[BOOST_BIND_ACTION(push_trim_type)] >> arithmetic_expression >> ')') [BOOST_BIND_ACTION(push_trim_expr)];
```

Comment:
as discussed, it need to add the trim rule to higher rule which should accept it (trim rule).